### PR TITLE
Add helm chart

### PR DESCRIPTION
**Pull request by mhrabovcin**
This PR adds `helm` chart for kube-oidc-proxy.

### CODE REVIEW

File: charts/kube-oidc-proxy/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+appVersion: "1.0"
```
Comment (JoshVanL) [inline charts/kube-oidc-proxy/Chart.yaml]: Can we also add a ` helm lint --strict` to the Makefile to check in CI?

File: charts/kube-oidc-proxy/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (JoshVanL) [inline charts/kube-oidc-proxy/LICENSE]: Think this needs to be added?

**Comment by mhrabovcin** [comment]
Hi @JoshVanL thanks for the feedback. I've worked on your comments. The main feature is that all `kube-oidc-proxy` flags have been exposed.

File: deploy/charts/kube-oidc-proxy/values.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  groupsClaim:
+  groupsPrefix:
+
+  signingAlgs:
+    - RS256
+  requiredClaims: {}
+  apiAudiences: []
+
+# To enable token passthrough feature
+# https://github.com/jetstack/kube-oidc-proxy/blob/feb9915fc5683bd1c3ff2bd0f393cb1f14aee822/docs/tasks/token-passthrough.md
```
Comment (JoshVanL) [inline deploy/charts/kube-oidc-proxy/values.yaml]: Can you change the SHA hash to `master`?

**Comment by mhrabovcin** [comment]
```
# FAILED!
# Cloning jetstack/kube-oidc-proxy at master(feb9915fc5683bd1c3ff2bd0f393cb1f14aee822)
# Checking out pulls:
#	97(2bcde686194ace198655bba7be142dfbcd03449b)
$ mkdir -p /home/prow/go/src/github.com/jetstack/kube-oidc-proxy
$ git init
Initialized empty Git repository in /home/prow/go/src/github.com/jetstack/kube-oidc-proxy/.git/
$ git config user.name ci-robot
$ git config user.email ci-robot@k8s.io
$ git fetch git@github.com:jetstack/kube-oidc-proxy.git --tags --prune
ssh: Could not resolve hostname github.com: Try again
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
# Error: exit status 128
```

Seems like DNS flake, I am going to restart the test.

**Comment by JoshVanL** [comment]
@mhrabovcin indeed, has been a bit flaky lately with DNS.

Cheers! :) 

/lgtm
/approve

**Pull request merged**